pageserver/src/tenant/config.rs

```diff
@@ -280,6 +282,179 @@ impl Default for TenantConf {
     }
 }
 
+// Helper function to standardize the error messages we produce on bad durations
+//
+// Intended to be used with anyhow's `with_context`, e.g.:
+//
+//   let value = result.with_context(bad_duration("name", &value))?;
+//
+fn bad_duration<'a>(field_name: &'static str, value: &'a str) -> impl 'a + Fn() -> String {
+    move || format!("Cannot parse `{field_name}` duration {value:?}")
+}
+
+impl TryFrom<&'_ TenantCreateRequest> for TenantConfOpt {
+    type Error = anyhow::Error;
```

Comment:
TenantCreateRequest essentially holds a TenantConfigRequest 
why not model it this way and we will have only one function for the parsing

Reply:
Had the same though, but, that would require manually checking that the two `from` impls in this PR do the same. 
Little bit riskier than this PR.
Let's do that in a separate PR?